modules/gapi/include/opencv2/gapi/util/variant.hpp

```diff
@@ -428,7 +518,152 @@ namespace util
     {
         return !(lhs == rhs);
     }
-} // namespace cv
+
+namespace detail
+{
+    // terminate recursion implementation for `non-void` ReturnType
+    template<typename ReturnType, std::size_t CurIndex, std::size_t ElemCount,
+             typename Visitor, typename Variant, typename... VisitorArgs>
+    ReturnType apply_visitor_impl(Visitor&& visitor,
+                                                     Variant& v,
+                                                     std::true_type processed,
+                                                     std::false_type no_return,
+                                                     VisitorArgs&& ...args)
+    {
+        // non-used params warning suppression
+        constexpr size_t non_variadic_args_num = 4;
+        std::array<bool, non_variadic_args_num + sizeof...(VisitorArgs)> dummy{
+                                    (suppress_unused_warning(visitor), true),
+                                    (suppress_unused_warning(v), true),
+                                    (suppress_unused_warning(processed), true),
+                                    (suppress_unused_warning(no_return), true),
+                                    (suppress_unused_warning(args),true)...};
```

Comment:
I don't quite get what it does - why do we need a construct like this? Does it have any side effects? I believe no?

Reply:
just stack allocation of several booleans in the end of recursion.

Used cause several compilers notifies it as `unused variable`

Reply:
So this whole construct is solely about the unused warning?

Reply:
I believe we can drop names from the method signature then and that's it :)